test/relations_test.dart

```diff
@@ -0,0 +1,146 @@
+import 'dart:typed_data';
+
+import 'package:test/test.dart';
+
+import 'entity.dart';
+import 'objectbox.g.dart';
+import 'test_env.dart';
+
+void main() {
+  /*late final*/ TestEnv env;
+
+  setUp(() {
+    env = TestEnv('relations');
+  });
+
+  tearDown(() {
+    env?.close();
+  });
+
+  test('to-one put', () {
+    final src = TestEntity(tString: 'Hello');
+    src.relA.attach(env.store);
+
+    expect(src.relA.hasValue, isFalse);
+    expect(src.relA.target, isNull);
+    src.relA.target = RelatedEntityA(tInt: 42);
+    expect(src.relA.hasValue, isTrue);
+    expect(src.relA.target, isNotNull);
+    expect(src.relA.targetId, isZero);
+    expect(src.relA.target.tInt, 42);
+
+    src.relA.target.relB.attach(env.store);
+    src.relA.target.relB.target = RelatedEntityB(tString: 'B1');
+
+    // TODO wait for #62, now duplicates the object without object ID assignment
+    // src.relB.target = src.relA.target.relB.target;
+
+    env.box.put(src);
+
+    var read = env.box.get(1);
+    expect(read.tString, equals(src.tString));
+    expect(read.relA.hasValue, isTrue);
+    expect(read.relA.targetId, 1);
+    expect(read.relA.target.tInt, 42);
+    expect(read.relA.target.relB.target.tString, equals('B1'));
+
+    // attach an existing item
+    expect(read.relA.target.relB.target.relA.hasValue, isFalse);
```

Comment:
Maybe refactor `read.relA.target.relB.target` into a variable? I have trouble reading this :)

Reply:
OK

Reply:
LGTM!